credsweeper/deep_scanner/deep_scanner.py

```diff
@@ -43,6 +44,17 @@ def __init__(self, config: Config, scanner: Scanner) -> None:
         """
         self.__config = config
         self.__scanner = scanner
+        self.__keywords = set()
+        for keyword_rule in [x[0] for x in scanner.rules_scanners if RuleType.KEYWORD == x[0].rule_type]:
+            self.__keywords.update(set(keyword_rule.required_substrings))
```

Comment:
As you know `Rule` class already has `required_substrings` and above logic of updating `__keywords` set seems duplicated and not that intuitive.
Is it impossible to change checking `required_substrings` by using `Rule` class?

Reply:
Probably i can rename self.__keywords to self.__all_rules_required_substrings ?
Iteration for rules requires more time than for the set.

Reply:
I mean the structure of each feature(`DeepScanner`, `Rule`, and so on) and its role can be mixed..
Actually each `required_substrings` is related with its `Rule` as you know..
So it is hard to agree with merging all `required_substrings` into one variable `__keywords`.

Reply:
Yes, each rule has separated required_strings list to trigger.
In html_representation we need to know pass the header to targets or not. In future the target will be scanned with rule as usually.
I'll do refactoring ...

Reply:
As you told, `represent_as_html()` method's role is represent the data to html not to check keywords and ignore or not.
After representation, `Rule` class will be used to filter the target is valid or not.
Please mind single responsibility principle.